neighsyncd/neighsync.cpp

```diff
@@ -9,14 +9,273 @@
 #include "ipaddress.h"
 #include "netmsg.h"
 #include "linkcache.h"
+#include "select.h"
 
 #include "neighsync.h"
+#include "warm_restart.h"
 
 using namespace std;
 using namespace swss;
 
-NeighSync::NeighSync(DBConnector *db) :
-    m_neighTable(db, APP_NEIGH_TABLE_NAME)
+const AppRestartAssist::cache_state_map AppRestartAssist::cacheStateMap =
+{
+    {STALE,     "STALE"},
+    {SAME,      "SAME"},
+    {NEW,       "NEW"},
+    {DELETE,    "DELETE"},
+    {UNKNOWN,   "UNKNOWN"}
+};
+
+AppRestartAssist::AppRestartAssist(RedisPipeline *pipelineAppDB,
+    const std::string &app_name, const std::string &docker_name,
+    ProducerStateTable *ps_table, const uint32_t defaultWarmStartTimerValue):
+    m_appTable(pipelineAppDB, APP_NEIGH_TABLE_NAME, false),
+    m_appName(app_name),
+    m_dockerName(docker_name),
+    m_psTable(ps_table),
+    warmStartTimer(timespec{0, 0})
+{
+    WarmStart::checkWarmStart(m_appName, m_dockerName);
+
+    m_appTableName = m_appTable.getTableName();
+
+    // set the default timer value
+    if (defaultWarmStartTimerValue != 0)
```

Comment:
>if (defaultWarmStartTimerValue != 0) [](start = 4, length = 36)

if (defaultWarmStartTimerValue != 0)
then m_reconcileTimer is uninitialized.

Reply:
This was already addressed in the new commit where there is a internal default timer.